docs: update styleguidist settings

uselessdev opened the pull request:
<!-- IMPORTANT: Please review the CONTRIBUTING.md file for detailed contributing guidelines and remove the items which you're not using. -->

## Context
This PR update the styleguidist settings, actually styleguidist is a quite verbose to configure.

## How test
1. Checkout to this branch: `git checkout update/styleguidist-settings`,
2. Run yarn to update the React Styleguidist dependency: `yarn`,
3. Run the styleguidist locally: `yarn styleguide`.

**To test a production version:**
You can use a library like [serve](https://www.npmjs.com/package/serve):

1. build the styleguide: `yarn styleguide:build`
2. start a serve: `serve -s config/styleguide`

Resolve #222

uselessdev commented on config/styleguide.config.js:
```diff
@@ -9,162 +8,28 @@ webpackConfig.module.rules.unshift({
+      usageMode: 'expand',
     },
   ],
-  ignore: [
```
Nope, because react-styleguidist [ignore tests files by default](https://react-styleguidist.js.org/docs/cookbook.html#how-to-exclude-some-components-from-style-guide) and we don't have any folder named as `tests` (i guess)

GitCop commented:
There were the following issues with your Pull Request

 * Commit: 87eaa0d52ab7392bb8dddd528cb7b0f743d70c6f
  * lways use the standard `scope: action description`, sempre em inglês  
    - The **scope** shouldn't have spaces. If the word is composite, use dashes.  
    - The **action** should be **add**, **remove**, **refactor**, **move**,  **bump**, **update** or **release**  
    - The **description** should contain a quick description of the change

The complete commit style guide is available at https://github.com/pagarme/git-style-guide

---
This message was auto-generated by https://gitcop.com

GitCop commented:
There were the following issues with your Pull Request

 * Commit: 87eaa0d52ab7392bb8dddd528cb7b0f743d70c6f
  * lways use the standard `scope: action description`, sempre em inglês  
    - The **scope** shouldn't have spaces. If the word is composite, use dashes.  
    - The **action** should be **add**, **remove**, **refactor**, **move**,  **bump**, **update** or **release**  
    - The **description** should contain a quick description of the change

The complete commit style guide is available at https://github.com/pagarme/git-style-guide

---
This message was auto-generated by https://gitcop.com

miguelssrs commented on config/styleguide.config.js:
```diff
@@ -7,29 +8,167 @@ webpackConfig.module.rules.unshift({
+      sections: [
+        {
+          name: 'Card',
+          content: '../src/Card/README.md',
```
Esses `content` estão sem exemplos no momento, né? testei clicar para isolar alguns desses componentes que agregam outros subcomponentes da mesma categoria e eles retornam um modelo de README da própria lib. Acho que esse é o comportamento esperado, certo?
![feb-15-2019 14-57-45](https://user-images.githubusercontent.com/20358128/52871945-5c68ab00-3132-11e9-9df6-a3f7653b486b.gif)

uselessdev replied:
Se não exitir exemplo no arquivo em questão, sim, talvez pode ser conflito no id da página ou algo assim 🤔 .

miguelssrs replied:
🤔

rsmelo commented on config/styleguide.config.js:
```diff
@@ -7,29 +8,167 @@ webpackConfig.module.rules.unshift({
   styleguideComponents: {
-    Wrapper: path.join(__dirname, './styleguide.js'),
+    Logo: path.join(__dirname, '../styleguide/components/Logo'),
+    // SectionRenderer: path.join(__dirname, '../styleguide/components/Section'),
```
pode remover essa linha comentada, as outras coisas tá show

rsmelo commented on styleguide/components/Section.js:
```diff
@@ -0,0 +1,41 @@
+import React from 'react'
+import SectionHeading from 'rsg-components/SectionHeading'
```
precisa adicionar essa dependencia no package.json?

uselessdev replied:
não, essa dependencia vem do `react-styleguidist`

rsmelo commented on styleguide/components/Section.js:
```diff
@@ -0,0 +1,41 @@
+
+import style from './style.css'
+
+export default (props) =>  {
```
```suggestion
export default (props) => {
```

rsmelo replied:
so percebi agora que esse componente não está sendo atribuído a uma variável ou nomeado, pode ficar ruim de debugar.
seria interessante:
```js
const Section = (props) => {.....
.....
export default Section
```

uselessdev replied:
Corrigido 😊

rsmelo commented on styleguide/components/Section.js:
```diff
@@ -0,0 +1,41 @@
+
+export default (props) =>  {
+  const {
+    content,
```
tá faltando definição de propTypes

rsmelo commented on styleguide/components/style.css:
```diff
@@ -0,0 +1,47 @@
+  background-color: #fefefe;
+}
+
+[data-preview] + div {
```
os seletores `data-preview] + div`, `data-preview] + div + div` e `data-preview] + div button` estão apresentando warning no console quando roda `yarn styleguide`

miguelssrs replied:
I saw this too, but I thought that this `[data-preview]` could be important. Maybe the solution is to disable style lint in these lines.

uselessdev commented:
@alalbux I fixed the spaces in the title and the order in the sidebar, however I did not change the logo in the introduction page because I think this page could be replaced by another page suggested by the UX team, I added this `introduction` page just as an example.

miguelssrs commented on styleguide/components/style.css:
```diff
@@ -0,0 +1,56 @@
+  background-color: #fefefe;
+}
+
+/* stylelint-disable selector-max-type */
```
👏
